valhalla/baldr/graphreader.h

```diff
@@ -343,6 +343,12 @@ class GraphReader {
   explicit GraphReader(const boost::property_tree::ptree& pt,
                        std::unique_ptr<tile_getter_t>&& tile_getter = nullptr);
 
+  void set_tile_getter_interrupt(const tile_getter_t::interrupt_t* interrupt) {
+    if (tile_getter_) {
+      tile_getter_->set_interrupt(interrupt);
```

Comment:
what is the process of cancelling everything from a higher level?
suppose we are building a route with thor which issues a blocking tile download.
if we want to abort all the activity, do we wrap always 'true' function and set it with set_tile_getter_interrupt?

Reply:
As I see it it will be something like 
```cpp
class WorkerWrapper {
public:
   WorkerWrapper() {
      interrupt_ = [this] (string) { return shouldInterrupt_; };
   }

  ~WorkerWrapper() {
      shouldInterrupt_ = {true};
   }

   void setupWorkers() {
      GraphReader reader;
      reader->set_interrupt(&interrupt_);
      boost::property_tree::ptree& conf;
      worker_ = make_unique<service_worker_t>(conf, reader);
   }
   
   void doWork() {
      try {
         worker_->route("");
      } catch (...) {
         // if we cancelled the fetch request we should fall here
      }
   }
  
private:
   atomic<bool> shouldInterrupt_{false};
   function<bool(string)> interrupt;
   unique_ptr<service_worker_t> worker_;
};
```

Reply:
Yeah this was my main question all along. At the moment the top level interrupts for the various workers can only be set via the http interface. we'd need to allow them being set separately and we'd need to also some how tile all the outstanding tile fetches to a particular request (in the case of multi threaded workers). i had talked to @vmihaylenko a few days ago about exactly this problem and i think its ok to solve this in a separate PR.

in summary and more specifically, id like to honor the existing interrupt structures we have for the top level worker actions and upon them triggering, id like to cascade that interrupt down to any in progress tile fetching....

actually i might have an idea here that might be a bit of a re-write of this work. let me see if i can formulate it into something actionable.